### PR TITLE
Some Cleanups

### PR DESCRIPTION
**Pull request by phimuemue**
I skimmed through the code this weekend and found some cleanup opportunities.

### CODE REVIEW

File: src/graphql.rs [hunk 1]
```diff
@@ -241,9 +241,9 @@ impl Revision {
         let commit = transaction.repo().find_commit(self.commit_id)?;
 
         let warnings = filter::compute_warnings(&transaction, self.filter, commit.tree()?)
-            .iter()
-            .map(|warn| Warning {
-                text: warn.to_string(),
+            .into_iter()
```
Comment (LMG) [inline src/graphql.rs:238]: Out of curiosity, why is it better?

Comment (phimuemue) [reply src/graphql.rs:238]: It avoids cloning the strings. (`iter` yields borrowed values that have to be (essentially) cloned via `to_string`, whereas `into_iter` allows for cheap moves.)

Comment (LMG) [reply src/graphql.rs:238]: thanks!

**Review by christian-schilling — CHANGES_REQUESTED** [review]
@phimuemue Thank you. There are some really good improvements in there.
Now it just needs to be buildable again 😅

**Comment by phimuemue** [comment]
> @phimuemue Thank you. There are some really good improvements in there.
> Now it just needs to be buildable again sweat_smile

Ouch! I guess I've run `cargo check` from the wrong directory.

**Review by christian-schilling — APPROVED** [review]

**Pull request merged**